src/psbt.test.js

```diff
@@ -39,6 +48,110 @@ describe("psbt", () => {
+        {},
+        {
+          xfp: ROOT_FINGERPRINT,
+          root: "m/45'/1'/100'",
+        },)).toBeNull();
+    })
+
+    it("should throw on psbt missing details (include psbt for another tx)", () => {
+      expect(() => {
+        translatePSBT(
+          tx.network,
+          tx.format,
+          TRANSACTIONS[1].psbt,
+          {
+            xfp: ROOT_FINGERPRINT,
+            root: "m/45'/1'/100'",
+          },
+        );
+      }).toThrow(/keyDetails not included/i);
+    })
+
+  });
+
+  describe("addSignaturesToPSBT", () => {
+    // Function expects Buffers but our fixtures are hex strings
+    const signaturesAsBuffers = tx.signature.map(sig => Buffer.from(sig, "hex"));
+    const signingPubKeyBuffer = Buffer.from(ms.publicKey, "hex");
+    const pubKeys = Array(signaturesAsBuffers.length).fill(signingPubKeyBuffer);
+
+    it("adds signatures to unsigned PSBT", () => {
+      const psbtWithSignature = addSignaturesToPSBT(tx.network, tx.psbt, pubKeys, signaturesAsBuffers);
+
+      expect(psbtWithSignature).toEqual(ms.psbtOrderedPartiallySigned);
+    });
+
+    it("throws validation error when trying to add valid signatures to the wrong pubkey", () => {
+      const wrongPubKeys = Array(signaturesAsBuffers.length).fill(Buffer.from(ms.publicKeys[1], "hex"));
+
+      expect(() => addSignaturesToPSBT(tx.network, tx.psbt, wrongPubKeys, signaturesAsBuffers)).toThrow(/invalid signature/i);
+    });
+
+    it("should return null with non-string PSBT input", () => {
+      expect(addSignaturesToPSBT(tx.network, [], pubKeys, signaturesAsBuffers)).toBeNull();
+    })
+  });
+
   describe("parseSignaturesFromPSBT", () => {
 
     const singleInputHexPSBT = '70736274ff0100530100000001656098cb3ee5c14be28c4cb8bf622520bbf9716846025677c0417f2c5ceca9da0000000000ffffffff01b3bff5050000000017a91420fd9addb8034a12ce353b8aa7f4faca50259b198700000000000100f702000000000101ba408eed9f0758c35eca8c0ef8e29cb63eed4c4d1b1349c989ce681f4f398f270000000017160014aba867b774169ca1e1bf635c204e3a4a80d7d0f9feffffff0200e1f5050000000017a914f745b90f312a38249a1e05c5db3a2f8d0aca4fdd87a02a29150100000017a9141d8a90a10f4ff666b36e797a4d00df9d36bd213c8702473044022004593ebe490574537724328a73b206863aea72b6b51b677e0aee6f7aa2afe54f0220560bd4a3665e34d559e96f77f6e6b619d214b36df04bcb58ffe0d74174cbd8e601210208741a0d458e76b355d1c274d78d8c165e207be0a5ff50623c0d63eb7932fa2f00000000010469522102315cbe4eada106463e04c238ffa3dbf26a77ad5047ac769c43f4629525266f4521029e866b92b02c686f89b320c0ce4c2391ee531b4dccfd719aefc8caa82e42c8282103ba8cb8627165090c2dca871274760b1ef027d89c6f54eb076e567a4f6c5e850853ae220603ba8cb8627165090c2dca871274760b1ef027d89c6f54eb076e567a4f6c5e85081cefa5d9162d00008001000080000000800100000000000000000000002206029e866b92b02c686f89b320c0ce4c2391ee531b4dccfd719aefc8caa82e42c8281cf57ec65d2d0000800100008032000080000000000000000000000000220602315cbe4eada106463e04c238ffa3dbf26a77ad5047ac769c43f4629525266f451cf57ec65d2d000080010000803c0000800000000000000000000000000000';
```

Comment:
Obviously this isn't new, but is there a reason these were included in the test file and not fixtures?

Reply:
IIRC -- bc they're not related to the key material in the fixtures